### PR TITLE
Final checkin to make the module stable

### PR DESCRIPTION
**Pull request by richardpark-msft**
- Vendor the /x/ module. It's public, but not API stable, so we'll just vendor the one that we used.
- Scrub the error message that can come from the file based method so we don't disclose the user's home folder (and possibly their username).

### CODE REVIEW

File: devid_all_platforms.go [hunk 1]
```diff
@@ -48,7 +50,9 @@ func readWriteDeviceIDFile(dir string) (string, error) {
 		}
 
 		if err := os.WriteFile(filePath, []byte(deviceID), 0600); err != nil {
-			return "", err
+			// this error might have some user information in it (via the home folder)
```
Comment (weikanglim) [inline devid_all_platforms.go:53]: I wonder how you feel when implementing PII scrubbing. This feels "best-effort" in many ways, given that we might not always understand what *may* or *may not* have PII information in it.

From a usage perspective, in azd, we do not log any messages, only codes and error types that we can parse and understand.

File: devid_all_platforms.go [hunk 2]
```diff
@@ -48,7 +50,9 @@ func readWriteDeviceIDFile(dir string) (string, error) {
...
+			// so let's scrub that out and make it something non-identifying and generic.
+			return "", scrubError(dir, err)
```
Comment (weikanglim) [inline devid_all_platforms.go:55]: Would we want to apply `scrubError` to the error handling for `os.ReadFile` (L60) and `os.MkdirAll` (L37) here as well?

File: devid_all_platforms.go [hunk 3]
```diff
@@ -58,3 +62,14 @@ func readWriteDeviceIDFile(dir string) (string, error) {
 
 	return string(contents), nil
 }
+
+func scrubError(folder string, err error) error {
+	oldMessage := err.Error()
+	newMessage := strings.Replace(oldMessage, folder, "(HOME)", 1)
+
+	if newMessage != oldMessage {
+		return errors.New(newMessage)
```
Comment (weikanglim) [inline devid_all_platforms.go:71]: We'd lose out on the error chain here. Maybe we can do a `wrapError` that still allows for `Unwrap()`?

File: devid_all_platforms.go (hunk 1 above)
Comment (weikanglim) [reply devid_all_platforms.go:53]: We spoke offline about it briefly, we're doing this as best practice, which is fine. It doesn't dictate how the caller should handle this.

File: devid_all_platforms.go (hunk 3 above)
Comment (weikanglim) [reply devid_all_platforms.go:71]: This is fine since we don't expect the consumer to have any expectations on the error chain.

**Review by weikanglim — APPROVED** [review]

File: devid_all_platforms.go (hunk 2 above)
Comment (richardpark-msft) [reply devid_all_platforms.go:55]: Yes! I'll add scrubs() in all the places and avoid any potential dependency on callers to error types.

**Pull request merged**